failed: EntityPassivationSpec in nightly build

https://github.com/akka/akka/runs/4293593178?check_suite_focus=true
 
Failed on nightly build on the 23rd Nov, the day in which https://github.com/akka/akka/pull/30933 was merged. Some extra tweaking for CI needed? 

```scala
<-- [Passivation of least frequently used entities must passivate the least frequently used entities when the per-shard entity limit is reached] End of log messages of test that [Failed(java.lang.AssertionError: assertion failed: timeout (12 seconds) during expectMsgClass waiting for class akka.cluster.sharding.EntityPassivationSpec$Entity$Received)]
[info] - must passivate the least frequently used entities when the per-shard entity limit is reached *** FAILED *** (12 seconds, 428 milliseconds)
[info]   java.lang.AssertionError: assertion failed: timeout (12 seconds) during expectMsgClass waiting for class akka.cluster.sharding.EntityPassivationSpec$Entity$Received
[info]   at scala.Predef$.assert(Predef.scala:223)
[info]   at akka.testkit.TestKitBase.expectMsgClass_internal(TestKit.scala:571)
[info]   at akka.testkit.TestKitBase.expectMsgType(TestKit.scala:553)
[info]   at akka.testkit.TestKitBase.expectMsgType$(TestKit.scala:552)
[info]   at akka.testkit.TestKit.expectMsgType(TestKit.scala:973)
[info]   at akka.cluster.sharding.AbstractEntityPassivationSpec.expectReceived(EntityPassivationSpec.scala:125)
[info]   at akka.cluster.sharding.LeastFrequentlyUsedEntityPassivationSpec.$anonfun$new$32(EntityPassivationSpec.scala:486)
[info]   at akka.cluster.sharding.LeastFrequentlyUsedEntityPassivationSpec.$anonfun$new$32$adapted(EntityPassivationSpec.scala:485)
[info]   at scala.collection.immutable.List.foreach(List.scala:431)
[info]   at akka.cluster.sharding.LeastFrequentlyUsedEntityPassivationSpec.$anonfun$new$24(EntityPassivationSpec.scala:485)
[info]   at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
```

--- a/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala
+++ b/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala
@@ -807,7 +807,6 @@ private[akka] class Shard(
       case Active(_) =>
         if (verboseDebug)
           log.debug("{}: Request to start entity [{}] (Already started)", typeName, entityId)
-        passivationStrategy.entityTouched(entityId)
         ackTo.foreach(_ ! ShardRegion.StartEntityAck(entityId, shardId))
       case _: RememberingStart =>
         entities.rememberingStart(entityId, ackTo)
@@ -1053,8 +1052,9 @@ private[akka] class Shard(
             case Active(ref) =>
               if (verboseDebug)
                 log.debug("{}: Delivering message of type [{}] to [{}]", typeName, payload.getClass.getName, entityId)
-              passivationStrategy.entityTouched(entityId)
+              val entitiesToPassivate = passivationStrategy.entityTouched(entityId)
               ref.tell(payload, snd)
+              passivateEntities(entitiesToPassivate)
             case RememberingStart(_) | RememberingStop | Passivating(_) =>
               appendToMessageBuffer(entityId, msg, snd)
             case state @ (WaitingForRestart | RememberedButNotCreated) =>
@@ -1069,7 +1069,10 @@ private[akka] class Shard(
             case NoState =>
               if (!rememberEntities) {
                 // don't buffer if remember entities not enabled
-                getOrCreateEntity(entityId).tell(payload, snd)
+                val ref = getOrCreateEntity(entityId)
+                val entitiesToPassivate = passivationStrategy.entityTouched(entityId)
+                ref.tell(payload, snd)
+                passivateEntities(entitiesToPassivate)
               } else {
                 if (entities.pendingRememberedEntitiesExist()) {
                   // No actor running and write in progress for some other entity id (can only happen with remember entities enabled)
@@ -1110,8 +1113,6 @@ private[akka] class Shard(
         context.watchWith(a, EntityTerminated(a))
         log.debug("{}: Started entity [{}] with entity id [{}] in shard [{}]", typeName, a, id, shardId)
         entities.addEntity(id, a)
-        val entitiesToPassivate = passivationStrategy.entityCreated(id)
-        passivateEntities(entitiesToPassivate)
         entityCreated(id)
         a
     }

--- a/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EntityPassivationStrategy.scala
+++ b/akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EntityPassivationStrategy.scala
@@ -53,17 +53,11 @@ private[akka] sealed abstract class EntityPassivationStrategy {
   def shardsUpdated(activeShards: Int): PassivateEntities
 
   /**
-   * A new entity instance has been created, which can trigger passivation.
-   * @param id entity id for the new entity instance
-   * @return entities to passivate, when active capacity has been reached
-   */
-  def entityCreated(id: EntityId): PassivateEntities
-
-  /**
    * An entity instance has been touched. Recorded before message delivery.
    * @param id entity id for the touched entity instance
+   * @return entities to passivate, when active capacity has been reached
    */
-  def entityTouched(id: EntityId): Unit
+  def entityTouched(id: EntityId): PassivateEntities
 
   /**
    * An entity instance has been terminated and should be removed from active tracking.
@@ -92,8 +86,7 @@ private[akka] object DisabledEntityPassivationStrategy extends EntityPassivation
   import EntityPassivationStrategy.PassivateEntities
 
   override def shardsUpdated(activeShards: Int): PassivateEntities = PassivateEntities.none
-  override def entityCreated(id: EntityId): PassivateEntities = PassivateEntities.none
-  override def entityTouched(id: EntityId): Unit = ()
+  override def entityTouched(id: EntityId): PassivateEntities = PassivateEntities.none
   override def entityTerminated(id: EntityId): Unit = ()
   override def scheduledInterval: Option[FiniteDuration] = None
   override def intervalPassed(): PassivateEntities = PassivateEntities.none
@@ -117,12 +110,10 @@ private[akka] final class IdleEntityPassivationStrategy(idleCheck: IdleCheck) ex
 
   override def shardsUpdated(activeShards: Int): PassivateEntities = PassivateEntities.none
 
-  override def entityCreated(id: EntityId): PassivateEntities = {
+  override def entityTouched(id: EntityId): PassivateEntities = {
     recencyList.update(id)
     PassivateEntities.none
   }
-
-  override def entityTouched(id: EntityId): Unit = recencyList.update(id)
 
   override def entityTerminated(id: EntityId): Unit = recencyList.remove(id)
 
@@ -151,12 +142,10 @@ private[akka] final class LeastRecentlyUsedEntityPassivationStrategy(perRegionLi
     passivateExcessEntities()
   }
 
-  override def entityCreated(id: EntityId): PassivateEntities = {
+  override def entityTouched(id: EntityId): PassivateEntities = {
     recencyList.update(id)
     passivateExcessEntities()
   }
-
-  override def entityTouched(id: EntityId): Unit = recencyList.update(id)
 
   override def entityTerminated(id: EntityId): Unit = recencyList.remove(id)
 
@@ -192,12 +181,10 @@ private[akka] final class MostRecentlyUsedEntityPassivationStrategy(perRegionLim
     passivateExcessEntities()
   }
 
-  override def entityCreated(id: EntityId): PassivateEntities = {
+  override def entityTouched(id: EntityId): PassivateEntities = {
     recencyList.update(id)
     passivateExcessEntities(skip = 1) // remove most recent before adding this created entity
   }
-
-  override def entityTouched(id: EntityId): Unit = recencyList.update(id)
 
   override def entityTerminated(id: EntityId): Unit = recencyList.remove(id)
 
@@ -237,12 +224,10 @@ private[akka] final class LeastFrequentlyUsedEntityPassivationStrategy(
     passivateExcessEntities()
   }
 
-  override def entityCreated(id: EntityId): PassivateEntities = {
+  override def entityTouched(id: EntityId): PassivateEntities = {
     frequencyList.update(id)
     passivateExcessEntities(skip = OptionVal.Some(id)) // make sure the newly created entity is still active
   }
-
-  override def entityTouched(id: EntityId): Unit = frequencyList.update(id)
 
   override def entityTerminated(id: EntityId): Unit = frequencyList.remove(id)
 

--- a/akka-cluster-sharding/src/test/scala/akka/cluster/sharding/passivation/simulator/Simulator.scala
+++ b/akka-cluster-sharding/src/test/scala/akka/cluster/sharding/passivation/simulator/Simulator.scala
@@ -211,7 +211,7 @@ object Simulator {
         Nil
       } else {
         activeEntities += entityId
-        val passivateEntities = strategy.entityCreated(entityId)
+        val passivateEntities = strategy.entityTouched(entityId)
         passivateEntities.foreach(activeEntities.remove)
         val passivated =
           if (passivateEntities.isEmpty) Nil

